tox.ini

```diff
@@ -110,6 +112,16 @@ deps = -r{toxinidir}/lint-requirements.txt
 skip_install = true
 usedevelop = false
 
+[testenv:coverage]
+description = Generate HTML coverage report
+deps =
+    -r{toxinidir}/agent/requirements.txt
+    -r{toxinidir}/server/test-requirements.txt
+commands =
+   coverage erase
+   pytest --cov --cov-append --cov-report=term-missing .
```

Comment:
Can we add these to `py3-server` and `py3-agent`?

If we are going to have a separate `tox` environment, can you run the agent and the server tests explicitly?  And fix the requirements?

Reply:
Because you wouldnt want to gate on python coverage because there is alot of coverage missing. When we get to that point then we can think of doing that.

Reply:
We can't just REPORT it without GATING success on it?

Reply:
I just tried the following change, and the py3-server tox environment passes, but the py3-agent fails because the tests are actually broken (command-line arguments of pytest are being used by the test itself).

The change below dumps the coverage reports into separate files:
```
[pportant@gandalf pbench]$ git diff
     1  diff --git a/tox.ini b/tox.ini
     2  index 07ed8ea89..40634f77a 100644
     3  --- a/tox.ini
     4  +++ b/tox.ini
     5  @@ -87,9 +87,13 @@ deps =
     6       -r{toxinidir}/server/requirements.txt
     7       -r{toxinidir}/server/test-requirements.txt
     8   commands =
     9  -    pytest ./lib/pbench/test/unit/common
    10  -    pytest ./lib/pbench/test/unit/server
    11  -    pytest ./lib/pbench/test/functional
    12  +    pytest --cov --cov-append \
    13  +      --cov-report html:cov_html \
    14  +      --cov-report xml:cov.xml \
    15  +      --cov-report annotate:cov_annotated \
    16  +      ./lib/pbench/test/unit/common \
    17  +      ./lib/pbench/test/unit/server \
    18  +      ./lib/pbench/test/functional
    19
    20   [testenv:py3-agent]
    21   description = Runs all Python3-based agent unit and functional tests
    22  @@ -97,9 +101,13 @@ deps =
    23       -r{toxinidir}/agent/requirements.txt
    24       -r{toxinidir}/agent/test-requirements.txt
    25   commands =
    26  -    pytest ./lib/pbench/test/unit/common
    27  -    pytest ./lib/pbench/test/unit/agent
    28  -    pytest -s ./lib/pbench/test/functional
    29  +    pytest --cov --cov-append \
    30  +      --cov-report html:cov_html \
    31  +      --cov-report xml:cov.xml \
    32  +      --cov-report annotate:cov_annotated \
    33  +      ./lib/pbench/test/unit/common \
    34  +      ./lib/pbench/test/unit/agent \
    35  +      ./lib/pbench/test/functional
    36
    37   [testenv:lint]
    38   description = Runs all linting tasks
```